From 1ad5e3cd6b8147d6919a2dbee9b905f86ba50c97 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Tue, 16 Jun 2020 12:31:36 +0200
Subject: [PATCH] [llvm] Disable linking llvm-exegesis to dylib

Force linking llvm-exegesis to static LLVM libraries instead of dylib
to prevent duplicate symbols due to linking both.  Ideally, we'd want
to link to the dylib only here but the target sub-libraries use hidden
symbols from LLVM target libraries and therefore linking the dylib
fails.

Differential Revision: https://reviews.llvm.org/D81922
---
 tools/llvm-exegesis/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/llvm-exegesis/CMakeLists.txt b/tools/llvm-exegesis/CMakeLists.txt
index a59e1b74024e..0575f2a06bb7 100644
--- a/tools/llvm-exegesis/CMakeLists.txt
+++ b/tools/llvm-exegesis/CMakeLists.txt
@@ -5,6 +5,7 @@ set(LLVM_LINK_COMPONENTS
   )

 add_llvm_tool(llvm-exegesis
+  DISABLE_LLVM_LINK_LLVM_DYLIB
   llvm-exegesis.cpp
   )
